File: monri/src/main/java/com/monri/android/ScandocUrlProvider.java
package com.monri.android;

class ScanDocUrlProvider {
    private final String baseUrl;
    private static final String BASE_URL_TERMINATION_CHAR = "/";
    private static final String AUTHENTICATION_ENDPOINT = "authenticate";
    private static final String AUTHENTICATION_REFRESH_ENDPOINT = "authenticate/refresh";
    private static final String EXTRACTION_ENDPOINT = "extraction";
    private static final String VALIDATION_ENDPOINT = "validation";

    protected ScanDocUrlProvider(final String baseUrl) {
        this.baseUrl = baseUrl.endsWith(BASE_URL_TERMINATION_CHAR) ? baseUrl : baseUrl + BASE_URL_TERMINATION_CHAR;
    }

    protected String getAuthenticationUrl() {
        return baseUrl + AUTHENTICATION_ENDPOINT;
    }

    protected String getExtractionUrl() {
        return baseUrl + EXTRACTION_ENDPOINT;
    }

    protected String getValidationUrl() {
        return baseUrl + VALIDATION_ENDPOINT;
    }

    protected String getAuthenticateRefreshUrl() {
        return baseUrl + getAuthenticateRefreshUrl();
    }
}

File: monri/src/main/java/com/monri/android/ScanDocUrlProvider.kt
package com.monri.android

internal class ScanDocUrlProvider(baseUrl: String): UrlProvider(baseUrl) {

    companion object {
        private const val AUTHENTICATION_BASE_URL = "https://api.scandoc.ai/ks/"
        private const val AUTHENTICATION_ENDPOINT = "authenticate/"
        private const val AUTHENTICATION_REFRESH_ENDPOINT = "authenticate/refresh"
        private const val EXTRACTION_ENDPOINT = "extraction/"
        private const val VALIDATION_ENDPOINT = "validation/"
    }

    val authenticationUrl: String
        get() = AUTHENTICATION_BASE_URL + AUTHENTICATION_ENDPOINT

    val extractionUrl: String
        get() = baseUrl + EXTRACTION_ENDPOINT

    val validationUrl: String
        get() = baseUrl + VALIDATION_ENDPOINT

    val authenticateRefreshUrl: String
        get() = AUTHENTICATION_BASE_URL + AUTHENTICATION_REFRESH_ENDPOINT
}
